src/agent/block_store_services/block_store_fs.js

```diff
@@ -133,7 +137,7 @@ class BlockStoreFs extends BlockStoreBase {
         let block_file;
 
         try {
-            if (config.BLOCK_STORE_FS_TMFS_ENABLED) {
+            if (!this.xattr_fallback && config.BLOCK_STORE_FS_TMFS_ENABLED) {
```

Comment:
I would leave this out - we don't plan to support it without xattr at the moment, so it would be best to fail if configured with tmfs and xattr is not supported rather than just ignoring the config.
```suggestion
            if (config.BLOCK_STORE_FS_TMFS_ENABLED) {
```